runtime/common/src/lib.rs

```diff
@@ -3,3 +3,33 @@
 pub mod constants;
 pub mod extensions;
 pub mod weights;
+
+/// Macro to set a value (e.g. when using the `parameter_types` macro) to either a production value
+/// or to an environment variable or testing value (in case the `frequency-rococo-local` feature is selected or in instant-sealing mode).
+/// Note that the environment variable is evaluated _at compile time_.
+///
+/// Usage:
+/// ```Rust
+/// parameter_types! {
+/// 	// Note that the env variable version parameter cannot be const.
+/// 	pub LaunchPeriod: BlockNumber = prod_or_local_or_env!(7 * DAYS, 1 * MINUTES, "FRQCY_LAUNCH_PERIOD");
+/// 	pub const VotingPeriod: BlockNumber = prod_or_local_or_env!(7 * DAYS, 1 * MINUTES);
+/// }
+/// ```
+#[macro_export]
+macro_rules! prod_or_local_or_env {
+	($prod:expr, $test:expr) => {
+		if cfg!(feature = "frequency-rococo-local") {
+			$test
+		} else {
+			$prod
+		}
+	};
+	($prod:expr, $test:expr, $env:expr) => {
+		if cfg!(feature = "frequency-rococo-local") {
+			core::option_env!($env).map(|s| s.parse().ok()).flatten().unwrap_or($test)
```

Comment:
I like this macro approach.
I think the name or documentation is a bit confusing about the priority of which value will be used in 3 param case. It looks like we first try to get the local env value and if not exists we use provided local value but it's not obvious that the logic is doing that without looking into the macro.

Reply:
Good question by @shannonwells , how this env variable will be set
Would we want to test different values locally without having to modify runtime code

Reply:
@aramikm I couldn't think of a good way to name the macro to get this grouping `prod OR ( env ELSE local )`. Open to naming thoughts.

Reply:
What about `use_prod_unless_local_then_env_check`?

Reply:
I would prefer to move the env to 2nd param and 3rd as local default and rename it as `use_prod_unless_local_env_or_default`